client/packages/common/src/hooks/useUrlQuery/useHandleUrlQueryParams.ts

```diff
@@ -0,0 +1,65 @@
+import { useUrlQuery } from './useUrlQuery';
+import { Column } from '@openmsupply-client/common';
+import { FilterController } from '../useQueryParams';
+
+// This hook uses the state of the url query parameters to provide query parameters and update methods to tables.
+
+export const useHandleUrlQueryParams = (filterIndex: string) => {
+  const { urlQuery, updateQuery } = useUrlQuery();
+
+  const updateSortQuery = (column: Column<any>) => {
+    const currentSort = urlQuery?.['sort'];
+    const sort = column.key as string;
+    if (sort !== currentSort) {
+      updateQuery({ sort, dir: '', page: '' });
+    } else {
+      const dir =
+        column?.sortBy?.direction === 'asc' || !column?.sortBy?.direction
+          ? 'desc'
+          : '';
+      updateQuery({ dir });
+    }
+  };
+
+  const updatePaginationQuery = (page: number) => {
+    // Page is zero-indexed in useQueryParams store, so increase it by one
+    updateQuery({ page: page + 1 });
+  };
+
+  const updateFilterQuery = (key: string, value: string) => {
+    updateQuery({ [key]: value });
+  };
+
+  const filter: FilterController = {
+    onChangeStringFilterRule: (key: string, _, value: string) =>
+      updateFilterQuery(key, value),
+    onChangeDateFilterRule: () => {},
+    onClearFilterRule: () => {},
+    filterBy: urlQuery?.[filterIndex]
+      ? {
+          [filterIndex]: { like: urlQuery?.[filterIndex] ?? '' },
+        }
+      : {},
+  };
+
+  const queryParams = {
+    page: urlQuery?.page ? urlQuery.page - 1 : 0,
+    offset: urlQuery?.page ? (urlQuery.page - 1) * 20 : 0,
+    first: 20,
```

Comment:
the magic number is reused twice here - would be great to have a CONST in order to provide some extra information about the intention and make it easy to change later..

```
const RECORDS_PER_PAGE = 20;
```

I like to put them at the top of page, after the imports, makes them easy to see when you first open the file.

```
    offset: urlQuery.page ? (urlQuery.page - 1) * RECORDS_PER_PAGE : 0,
    first: RECORDS_PER_PAGE,
```

Reply:
Yup, good idea. Do we have a global constants file anywhere? Should be start one? (Cos this value of 20 records per page is used elsewhere too)

Reply:
@mark-prins  I made a new issue if we want to start a global constants file: https://github.com/openmsupply/open-msupply/issues/42

Feel free to close if it's unnecessary at this point.

Reply:
no, we don't - that's a better idea. Not entirely sure where to put that though, this code is in common but I would rather have the records per page in an app-specific common area. Looks like I need to split out common sooner rather than later...